src/dtproject/assets/schemas/devcontainers.json

```diff
@@ -1,12 +1,66 @@
     "$comment": "Schema for /dtproject/devcontainers.yaml file",
+    "title": "Duckietown devcontainers",
     "type": "object",
     "description": "Dictionary of named devcontainer-compatible configurations",
     "patternProperties": {
         ".*": {
             "type": "object",
-            "description": "DevContainer-compatible configuration. This is whatever you would put inside a '.devcontainer.json' file"
+            "description": "DevContainer-compatible configuration. This is whatever you would put inside a '.devcontainer.json' file",
+            "properties": {
+                "dockerComposeFile": {
+                    "type": "string",
+                    "description": "The path to the docker-compose file to use for the devcontainer",
+                    "default": "docker-compose.yml"
+                },
+                "service": {
+                    "type": "string",
+                    "description": "The name of the service to use for the devcontainer",
+                    "default": "devcontainer"
+                },
+                "workspaceFolder": {
+                    "type": "string",
+                    "default": "/code/catkin_ws/src/${localWorkspaceFolderBasename}",
+                    "description": "Workspace folder path"
+                },
+                "customizations": {
+                    "type": "object",
+                    "properties": {
+                        "vscode": {
+                            "type": "object",
+                            "properties": {
+                                "extensions": {
+                                    "type": "array",
+                                    "default": [
+                                        "ms-iot.vscode-ros",
+                                        "ms-python.python",
+                                        "ms-vscode.cpptools"
+                                    ],
+                                    "items": {
+                                        "type": "string"
+                                    },
+                                    "description": "Array of Visual Studio Code extensions"
+                                }
+                            },
+                            "required": [
+                                "extensions"
+                            ]
+                        }
+                    },
+                    "default": {
+                        "vscode": {
+                            "extensions": [
+                                "ms-iot.vscode-ros",
+                                "ms-python.python",
+                                "ms-vscode.cpptools"
+                            ]
+                        }
+                    },
+                    "description": "Customizations for Visual Studio Code"
+                }
+            },
+            "required": ["dockerComposeFile", "service", "workspaceFolder"]
```

Comment:
The fields `"dockerComposeFile", "service", "workspaceFolder"` cannot be required while also having a default value. In fact, they should not be required and their default value should be `null`, signaling that the user did not specify a value otherwise it would be impossible to make it work with the `configurations.yaml` layer.